src/renderer/VisibleRenderer.ts

```diff
@@ -4,8 +4,62 @@
  */
 import Renderer from "./Renderer";
 
+import { getFlickingAttached } from "~/utils";
+import { Panel } from "~/core";
 class VisibleRenderer extends Renderer {
+  public render() {
+    const flicking = getFlickingAttached(this._flicking, "Renderer");
+    const camera = flicking.camera;
+    const cameraElement = camera.element;
+    const visiblePanels = camera.visiblePanels;
 
+    const fragment = document.createDocumentFragment();
+
+    const visibleSortedByActualPosition = [...visiblePanels]
+      .sort((a, b) => (a.position + a.offset) - (b.position + b.offset));
+
+    const offsetGtZero = this._panels.filter(panel => panel.offset > 0);
+    const visibleOffsetLteZero = visiblePanels.filter(panel => panel.offset <= 0);
+
+    visibleSortedByActualPosition.forEach(panel => {
+      fragment.appendChild(panel.element);
+    });
+
+    // Remove remaining(removed) elements
+    while (cameraElement.firstChild) {
```

Comment:
I think that`firstElementChild` seems logically more appropriate than `firstChild` with text nodes. And if you're going to delete all of them, it's a live DOM, so it looks better to process it with `innerHTML=""`. What do you think?

Reply:
I believe I used removing all the `firstChild` method as it's the fastest way for doing that while I'm researching for Flicking v3.
- https://www.measurethat.net/Benchmarks/Show/10677/0/remove-all-children-from-dom-element-with-lastchild#latest_results_block
Also you're right that `firstElementChild` is more logically appropriate, but as Flicking shouldn't have text nodes inside I think it is also correct to remove all the child nodes.